tests/JMS/Serializer/Tests/Serializer/XmlSerializationTest.php

```diff
@@ -361,6 +363,20 @@ public function testWithoutFormatedOutputByXmlSerializationVisitor()
         $this->assertXmlStringEqualsXmlString($this->getContent('simple_class_object_minified'), $stringXml);
     }
 
+    public function testDiscriminatorAsXmlAttribute()
+    {
+        $xml = simplexml_load_string($this->serialize(new ObjectWithXmlAttributeDiscriminatorChild()));
+        $this->assertEquals('type="child"', trim($xml->xpath('//result/@type')[0]->saveXML()));
```

Comment:
instead of "re-loading" the xml into a simplexml object, why not re-use the `$this->getContent('template_name')` method? the whole test case will be: 

```php
$xml = $this->serialize(new ObjectWithXmlAttributeDiscriminatorChild());
$this->assertEquals($this->getContent('xml_discriminator_attribute'), $xml);
$this->assertInstanceOf(
    ObjectWithXmlAttributeDiscriminatorChild::class,
    $this->deserialize(
         $xml,
        ObjectWithXmlAttributeDiscriminatorParent::class
     )
);
```